ui/src/app/views/conceptset-create-modal/component.tsx

```diff
@@ -0,0 +1,122 @@
+import {Component, Input} from '@angular/core';
+import {AlertDanger} from 'app/components/alert';
+import {Button} from 'app/components/buttons';
+import {TextInput, ValidationError} from 'app/components/inputs';
+import {Modal, ModalBody, ModalFooter, ModalTitle} from 'app/components/modals';
+import {WorkspaceData} from 'app/resolvers/workspace';
+import {conceptSetsApi} from 'app/services/swagger-fetch-clients';
+import {ReactWrapperBase, summarizeErrors, withCurrentWorkspace} from 'app/utils/index';
+import {ConceptSet, CreateConceptSetRequest, Domain, DomainInfo} from 'generated/fetch';
+import * as React from 'react';
+import {validate} from 'validate.js';
+
+
+export const CreateConceptSetModal = withCurrentWorkspace()
+(class extends React.Component<{
+  workspace: WorkspaceData,
+  onCreate: Function,
+  onClose: Function,
+  conceptDomainList: Array<DomainInfo>,
+  existingConceptSets: ConceptSet[],
+}, {
+  name: string,
+  description: string,
+  domain: Domain,
+  nameTouched: boolean,
+  saving: boolean,
+  savingError: string,
+}> {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      name: '',
+      description: '',
+      domain: props.conceptDomainList[0] as unknown as Domain,
+      nameTouched: false,
+      saving: false,
+      savingError: ''
+    };
+  }
+
+  async saveConceptSet() {
+    try {
+      const {name, description, domain} = this.state;
+      const {workspace} = this.props;
+      const concept: ConceptSet = {
+        name: name,
+        description: description,
+        domain: domain
+      };
+      const request: CreateConceptSetRequest = {
+        conceptSet: concept
+      };
```

Comment:
FYI, typescript can perform type inference in straightforward cases like this, so it will be able to verify the shape of the request object without being explicit. This could be shortened to e.g.
```
const request = {conceptSet: {name, description, domain}};
```